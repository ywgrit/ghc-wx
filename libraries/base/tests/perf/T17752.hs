module T17752 where

-- All occurrences of elem should be optimized away.
-- For strings these should result in loops after inlining foldCString.
-- For lists it should result in a case expression.

-- Should compile to a pattern match if the rules fire
isElemList    x = x `elem` ['a','b','c']
isNotElemList x = x `elem` ['x','y','z']

isOneOfThese x = x `elem` [1,2,3,4,5::Int]
isNotOneOfThese x = x `notElem` [1,2,3,4,5::Int]

isElemString x = elem x "foo"
isNotElemString x = notElem x "bar"

isElemStringUtf x = elem x "foö"
isNotElemStringUtf x = notElem x "bär"
