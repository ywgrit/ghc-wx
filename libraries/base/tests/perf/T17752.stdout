[1 of 1] Compiling T17752           ( T17752.hs, T17752.o )

